azure-kusto-ingest/azure/kusto/ingest/ingest_client.py

```diff
@@ -17,18 +19,27 @@
 from .ingestion_properties import DataFormat, IngestionProperties
 
 
-class KustoIngestClient:
+class QueuedIngestClient:
     """
-    Kusto ingest client provides methods to allow ingestion into kusto (ADX).
+    Kusto queued ingest client provides methods to allow ingestion into kusto (ADX).
     To learn more about the different types of ingestions and when to use each, visit:
     https://docs.microsoft.com/en-us/azure/data-explorer/ingest-data-overview#ingestion-methods
     """
 
+    _INGEST_PREFIX = "ingest-"
+    _WRONG_ENDPOINT_MESSAGE = (
+        "You are using '{0}' client type, but the provided endpoint is of ServiceType '{1}'. Initialize the client with the appropriate endpoint URI"
+    )
+    _EXPECTED_SERVICE_TYPE = "DataManagement"
+
     def __init__(self, kcsb: Union[str, KustoConnectionStringBuilder]):
         """Kusto Ingest Client constructor.
         :param kcsb: The connection string to initialize KustoClient.
         """
         self._resource_manager = _ResourceManager(KustoClient(kcsb))
+        if not isinstance(kcsb, KustoConnectionStringBuilder):
+            kcsb = KustoConnectionStringBuilder(kcsb)
```

Comment:
If you create KCSB anyway, you can pass it to the KustoClient above and don't create it twice (inside KustoClient)